bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,98 @@
+from asyncio import TimeoutError
+from pathlib import Path
+from random import choice
+
+from discord import Color, Embed
+from discord.ext import commands
+
+from bot.bot import Bot
+
+
+class Hangman(commands.Cog):
+    """
+    Cog for the Hangman game.
+
+    Hangman is a classic game where the user tries to guess a word, with a limited amount of tries.
+    """
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        with Path("bot/resources/fun/top_1000_used_words.txt").resolve().open(mode="r", encoding="utf-8") as file:
+            self.all_words = list(map(lambda line: line.strip("\n"), file.readlines()))
+
+    @commands.command()
+    async def hangman(self, ctx: commands.Context, min_length: str = "0", max_length: str = "25",
+                      min_unique_letters: str = "0", max_unique_letters: str = "25") -> None:
```

Comment:
PyDis usually uses the following style for too much parameters/too long parameters:
```py
def function(
    long_param_1: type_annotation = default_value,
    long_param_2: type_annotation = default_value,
    long_param_3: type_annotation = default_value,
) -> return_type:
```
So it might be best to stick to it for consistency. Note that this is not only for function definitions - any kind of enclosing expression/syntax adheres to the style above.